src/pytestqt/wait_signal.py

```diff
@@ -257,12 +259,12 @@ def _get_timeout_error_message(self):
             )
 
 
+@dataclasses.dataclass
 class SignalAndArgs:
-    def __init__(self, signal_name, args):
-        self.signal_name = signal_name
-        self.args = args
+    signal_name: str
+    args: list[Any]
```

Comment:
```suggestion
    args: list[object]
```

In this case, we don't care about the type of the object, for this reason `object` is safer: it will accept anything (just like `Any`), but contrary to `Any`, the type checker will flag if we accidentally try to perform some operation on it (say call a method).

Reply:
Yep, see https://pytest-qt.readthedocs.io/en/latest/signals.html#getting-emitted-signals-and-arguments:

> To determine which of the expected signals were emitted during a `wait()` you can use `blocker.all_signals_and_args` which contains a list of `wait_signal.SignalAndArgs` objects, indicating the signals (and their arguments) in the order they were received.

I don't think this can be made generic, as the arguments will depend on the signal being recorded, and we don't know which signals (and how many in total) there will be in there.

Good catch regarding `tuple[Any, ...]`, will adjust accordingly.